src/cloud_tasks/instance_manager/gcp.py

```
@@ -262,6 +272,87 @@ def _get_compute_client(self):
+                    data[key_str] = value
+            fd, temp_path = tempfile.mkstemp(dir=cache_dir or None, suffix=".json", prefix=".cloud_tasks_gcp_pricing_")
+            with os.fdopen(fd, "w") as f:
+                json.dump(data, f, indent=2)
+                f.flush()
+                try:
+                    os.fsync(f.fileno())
+                except OSError:
+                    pass
+            os.replace(temp_path, path)
+            temp_path = None
+            self._logger.debug(f"Saved GCP pricing cache to {path}")
```

Ruff (0.14.14)

[warning] Logging statement uses f-string

(G004)